src/snowflake/snowpark/types.py

```diff
@@ -284,6 +284,23 @@ def _to_attributes(self) -> List["expression.Attribute"]:
     def __repr__(self) -> str:
         return f"StructType([{', '.join(repr(f) for f in self.fields)}])"
 
+    def __getitem__(self, item: Union[str, int, slice]) -> StructField:
+        """Access fields by name, index or slice."""
+        if isinstance(item, str):
+            for field in self.fields:
+                if field.name == item:
+                    return field
+            raise KeyError(f"No StructField named {item}")
+        elif isinstance(item, int):
+            return self.fields[item]  # may throw ValueError
+        elif isinstance(item, slice):
+            return StructType(self.fields[item])
+        else:
+            raise TypeError("StructType items should be strings, integers or slices")
```

Comment:
nit: add at the end `, but got {type(item)}`

Reply:
Done